reserve-crate: Cannot install cargo-generate

I would want to reserve a substrate crate name using the git actions.

During the action: https://github.com/paritytech/releng-scripts/actions/runs/4566220209/jobs/8058412281 the following error was encountered:

```rust
cargo install cargo-generate --version 0.17.0
...
 error[E0277]: the trait bound `BStr: AsRef<[_; 0]>` is not satisfied
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.18.0/src/store/packed/find.rs:95:52
   |
95 |                 .map(|(_rest, r)| r.name.as_bstr().as_ref())
   |                                                    ^^^^^^ the trait `AsRef<[_; 0]>` is not implemented for `BStr`
   |
   = help: the following other types implement trait `AsRef<T>`:
             <BStr as AsRef<BStr>>
             <BStr as AsRef<[u8]>>

error[E0277]: the trait bound `BStr: AsRef<[_; 0]>` is not satisfied
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.18.0/src/store/packed/find.rs:91:43
   |
91 |         a.binary_search_by_key(&full_name.as_ref(), |b: &u8| {
   |                                           ^^^^^^ the trait `AsRef<[_; 0]>` is not implemented for `BStr`
   |
   = help: the following other types implement trait `AsRef<T>`:
```

--- a/.github/workflows/reserve-crate.yml
+++ b/.github/workflows/reserve-crate.yml
@@ -31,20 +31,22 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Install Rust
-        uses: dtolnay/rust-toolchain@55c7845fad90d0ae8b2e83715cb900e5e861e8cb # stable branch
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
         with:
+          profile: minimal
           toolchain: stable
+          override: true
 
       # Put this AFTER installing Rust because the cache uses the current rustc
       # version as its cache key
-      - name: Cache Rust
-        uses: Swatinem/rust-cache@359a70e43a0bb8a13953b04a90f76428b4959bb6 # v2.2.0
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f # v2.2.1
         with:
           cache-on-failure: "true"
 
       - name: Install cargo-generate
-        run: cargo install cargo-generate --version 0.17.0
+        run: cargo install cargo-generate --version 0.18.1
 
       - name: Reserve the crate
         env:

